application/yii2/Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM composer:1.10.10
+
+# Initialize Yii2 project
+RUN composer create-project --prefer-dist yiisoft/yii2-app-basic basic
```

Comment:
I see that you're creating a new Yii2 project in the container and then copying over the files that you modified for the example.

Would this be done in an actual Yii2 project?

I feel like a better approach would be to generate the project outside the container, in the examples repo and then copy all of it into the container. Then we wouldn't even need to use the composer base image and we could instead base off `php` or something else.

Let me know if I'm missing something or you disagree,

Reply:
I think that the way we're doing it at the moment is actually better, these are my considerations:
1. In the majority of the other examples, this is done exactly the same way, i.e. by creating a clean project and then copying the files that would usually need to be modified in the container.
2. We would still need to install composer, since the `php yii` command comes with composer.
3. This might be because of personal preference, but I think that it is clearer to the visitor of this example repo what they actually need to do in order to achieve the same effect that the example does, if we just highlight the differences between a clean project and a SecretHub example project. We're doing this at the moment by only moving the files that are affected by the change.

What do you think about this?